blog/2021-12-20-openhab-3-1-release.md

```diff
@@ -0,0 +1,79 @@
+
+This behavior is configurable, and users can opt to disable automatic imports into scripts for full control of their scripting environment.  Users can continue to use custom built libraries as well as NPM modules through the use of CommonJS Require.  Best of all, [openhab-js](https://github.com/openhab/openhab-js) is published as an NPM library, [openhab](https://www.npmjs.com/package/openhab) and can be updated and or modified by users independently of the add-on shipped version by running “npm -i openhab” inside the user’s JS automation configuration folder. 
+
+The JS Scripting binding also now includes standard built in global features that users coming from modern JavaScript environments expect to have.  This includes full support of [console logging](https://developer.mozilla.org/en-US/docs/Web/API/Console/log) to the main openHAB log (including debug, trace, info, etc.), timer support with “[setTimeout()](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout)” and “[setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/setInterval)” , and importing personal or NPM published packages with “require”.  For example, logging in rules is as easy as:
+
+```javascript
+console.log("Kitchen Light State", items.getItem("KitchenLight").state);
+```
+
+Finally, in previous openHAB versions, installing the JS Scripting binding meant the legacy default Javascript engine (Nashorn) could not be used.  In practice the two engines had issues coexisting, especially when it comes to UI based rules.  In 3.2 JS Scripting will use its own directory for files in “conf/automation/js, including user provided libraries.  The UI will now show both scripting engines when selecting an action, allowing the user to continue to use the legacy JavaScript engine, or migrate to this new ES6 based version at their convenience.  
+
+Thank you to Jonathan Gilbert (aka jpg0), who contributed both the JSScripting add-on as well as the openhab-js library. 
+
+## Marketplace 
+
+## openHAB Alexa Skill
+
+## Localizing Add-Ons
+
+Previous versions of openHAB have focused on localizing the openHAB Main UI as well as configuration settings in the core. In this release we have focused on setting up an easy to use translation toolbox for openHAB Add-Ons. Our new i18n-tool allows developers to create the default files containing English terms in their familiar Maven environment (just run `mvn i18n:generate-default-translations` in the bundle folder). Once added to GitHub those files will be uploaded automatically to [Crowdin](https://crowdin.com/project/openhab-addons) - a collaboration platform to support multilingual translations. This will allow everyone in our community to translate into their language of choice without requiring any development skills. Completed translations are synced back to our repository periodically. If your language is missing, please help and visit [Crowdin](https://crowdin.com/project/openhab-addons)!
```

Comment:
```suggestion
Previous versions of openHAB have focused on localizing the openHAB Main UI as well as configuration settings in the core. In this release we have focused on setting up an easy to use translation toolbox for openHAB Add-ons. Our new i18n-tool allows developers to create the default files containing English terms in their familiar Maven environment (just run `mvn i18n:generate-default-translations` in the bundle folder). Once added to GitHub those files will be uploaded automatically to [Crowdin](https://crowdin.com/project/openhab-addons) - a collaboration platform to support multilingual translations. This will allow everyone in our community to translate into their language of choice without requiring any development skills. Completed translations are synced back to our repository periodically. If your language is missing, please help and visit [Crowdin](https://crowdin.com/project/openhab-addons)!
```